app/src/main/java/nl/nfi/cellscanner/MainActivity.java

```diff
@@ -45,60 +43,32 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setContentView(nl.nfi.cellscanner.R.layout.activity_main);
 
         exportButton = findViewById(nl.nfi.cellscanner.R.id.exportButton);
         clearButton = findViewById(nl.nfi.cellscanner.R.id.clearButton);
-        recorderSwitch = findViewById(nl.nfi.cellscanner.R.id.recorderSwitch);
 
-        recorderSwitch.setChecked(Recorder.inRecordingState(getApplicationContext()));
```

Comment:
Is it set somewhere to checked/unchecked based on setting in previous state?

Reply:
Yes, this is statefull set in key value pairs